projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -27,6 +53,1244 @@ export interface IgxComboBase {
+     * ```
+     */
+    @Input()
+    public placeholder;
+
+    /**
+     * Defines the placeholder value for the combo dropdown search field
+     *
+     * ```typescript
+     * // get
+     * let myComboSearchPlaceholder = this.combo.searchPlaceholder;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [searchPlaceholder]='newPlaceHolder'></igx-combo>
+     * ```
+     */
+    @Input()
+    public searchPlaceholder = 'Enter a Search Term';
+
+    /**
+     * Defines whether the caseSensitive icon should be shown in the search input
+     *
+     * ```typescript
+     * // get
+     * let myComboShowSearchCaseIcon = this.combo.showSearchCaseIcon;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [showSearchCaseIcon]='true'></igx-combo>
+     * ```
+     */
+
+    @Input()
+    public showSearchCaseIcon = false;
+
+    /**
+     * Combo data source.
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [data]='items'></igx-combo>
+     * ```
+     */
+    @Input()
+    public get data(): any[] | null {
+        return this._data;
+    }
+    public set data(val: any[] | null) {
+        this._data = (val) ? val : [];
+    }
+
+    /**
+     * Combo value data source property.
+     *
+     * ```typescript
+     * // get
+     * let myComboValueKey = this.combo.valueKey;
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [valueKey]='myKey'></igx-combo>
+     * ```
+     */
+    @Input()
+    public valueKey: string = null;
+
+    @Input()
+    public set displayKey(val: string) {
+        this._displayKey = val;
+    }
+
+    /**
+     * Combo text data source property.
+     *
+     * ```typescript
+     * // get
+     * let myComboDisplayKey = this.combo.displayKey;
+     *
+     * // set
+     * this.combo.displayKey = 'val';
+     *
+     * ```
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [displayKey]='myDisplayKey'></igx-combo>
+     * ```
+     */
+    public get displayKey() {
+        return this._displayKey ? this._displayKey : this.valueKey;
+    }
+
+    /**
+     * The item property by which items should be grouped inside the items list. Not usable if data is not of type Object[].
+     *
+     * ```html
+     * <!--set-->
+     * <igx-combo [groupKey]='newGroupKey'></igx-combo>
+     * ```
+     */
+    @Input()
+    public set groupKey(val: string) {
+        this._groupKey = val;
+    }
+
+    /**
+     * The item property by which items should be grouped inside the items list. Not usable if data is not of type Object[].
+     *
+     * ```typescript
+     * // get
+     * let currentGroupKey = this.combo.groupKey;
+     * ```
+     */
+    public get groupKey(): string {
+        return this._groupKey;
+    }
+
+    /**
+     * An @Input property that enabled/disables filtering in the list. The default is `true`.
+     * ```html
+     * <igx-combo [filterable]="false">
+     * ```
+     */
+    @Input()
+    public filterable = true;
```

Comment:
do we need this for simple combo? Setting this to `false` there makes UX really terrible.